codec/src/test/java/io/netty/handler/codec/compression/BrotliDecoderTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.codec.compression;
+
+import com.aayushatharva.brotli4j.encoder.BrotliOutputStream;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.embedded.EmbeddedChannel;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.experimental.theories.DataPoints;
+import org.junit.experimental.theories.FromDataPoints;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Theories.class)
+public class BrotliDecoderTest {
```

Comment:
+1 to @normanmaurer's suggestion to create an issue, describing why `AbstractDecoderTest` could not be used with `BrotliDecoder`.

> Please hold reviews for now. I'm investigating a test failure I'm having a hard time reproducing locally. I have to stay the test suite is very weird, with  static `AbstractDecoderTest#compressedBytesLarge` actually loaded from ctor. This seems to be some JUnit hack, but this hack crashes with NPE when trying to run one single test, eg `mvn -Dtest=io.netty.handler.codec.compression.BrotliDecoderTest test`.

IIUC, it might be some visibility issue between jUnit threads

Reply:
https://github.com/netty/netty/issues/11233